custom_components/ble_monitor/ble_parser/__init__.py

```diff
@@ -84,8 +85,11 @@ def parse_data(self, data):
                     if uuid16 == 0xFFF9 or uuid16 == 0xFDCD:  # UUID16 = Cleargrass or Qingping
                         sensor_data = parse_qingping(self, adstruct, mac, rssi)
                         break
-                    elif uuid16 == 0x181A:  # UUID16 = ATC
-                        sensor_data = parse_atc(self, adstruct, mac, rssi)
+                    elif uuid16 == 0x181A:  # UUID16 = ATC or b-parasite
```

Comment:
This is a design decision in your firmware, using the ATC UUID16. At the moment, this will work, but if the ATC firmware devs decide to add an extra byte in the future, it might bite with your sensors (ATC already has 4 formats with different length). You might want to decide to change the UUID to something else in the future. UUID are normally reserved for actual companies, most "unofficial" BLE implementations use the format `FF` + 2 bytes as a company identifier, see e.g. the Govee sensors. I do understand if you don't change this now, but you might want to consider this for future development.

Reply:
0x181A is not allocated to ATC or any company. It rather is a generic UUID16 for a "Environmental Sensing" GATT Service. Thus any device may publish using that UUID, not only ATC (https://btprodspecificationrefs.blob.core.windows.net/assigned-values/16-bit%20UUID%20Numbers%20Document.pdf). Maybe there is some other identifier (besides mac address) usable here?

Reply:
Yes, I know, I just want to warn you that you might get into trouble if the ATC devs decide to make a 20 byte long advertisement. But it's ok to leave it like it is for now, and change it if the ATC devs decide to change the format in the future. I have no idea if they have any plans to do. 

Alternative is, in stead of `16181A` is using `FFxxyy`, where for `xxyy`, you choose something that is not in this list 

https://www.bluetooth.com/specifications/assigned-numbers/company-identifiers/

Reply:
I understand, sorry for sounding condescending. I am not the creator of the b-parasite device, but changing the UUID would break a lot of already deployed devices. This seems weird from a BLE-spec perspective, but maybe I'm missing something. 
If this breaks in the future, perhaps we can make use of the first two bytes of the data to differentiate between ATC and b-parasite (https://github.com/rbaron/b-parasite/tree/main/code/b-parasite#bluetooth-low-energy-advertisement-data-encoding)

Reply:
ah, I though you were the creator of these b-parasite device. As said before, no problem to leave it like it is. If ATC format changes, we can look into it how to solve.